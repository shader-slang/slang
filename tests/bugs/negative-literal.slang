//TEST(compute):COMPARE_COMPUTE_EX:-cpu -compute -compile-arg -O3 -shaderobj
//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj
//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -dx12 -shaderobj
//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name outputBuffer
RWStructuredBuffer<float> outputBuffer;

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    float v = asfloat((int(dispatchThreadID.x) & -2147483648) | 1065353216);
	outputBuffer[dispatchThreadID.x] = v;
}
