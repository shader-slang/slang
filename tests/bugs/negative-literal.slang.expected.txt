3F800000
3F800000
3F800000
3F800000
